api/src/main/java/org/pmiops/workbench/google/CloudStorageServiceImpl.java

```diff
@@ -30,6 +33,18 @@ public String readBlockscoreApiKey() {
     return readToString(getCredentialsBucketName(), "blockscore-api-key.txt").trim();
   }
 
+  @Override
+  public List<String> getBucketFileList(String bucketName, String directory) {
+    List<String> fileNameList = new ArrayList<>();
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+    Iterable<Blob> blobList = storage.get(bucketName)
+        .list(Storage.BlobListOption.prefix(directory)).getValues();
+    blobList.forEach(blobItem -> {
+      fileNameList.add(blobItem.getName());
```

Comment:
This name contains the path, e.g. notebooks/blah.ipynb?

Reply:
Yes. Updated to return BlobInfo per size() comment.